### PR TITLE
add new `unstable_mermaid` loader option for replacing '''mermaid  code-blocks with `<Mermaid />` (nextra/mermaid) React component

### PR DESCRIPTION
**Pull request by B2o5T**
In The Guild, we use a lot of rendering mermaid diagrams, but unfortunately, there is no plugin for MDX2 for replacing \`\`\`mermaid code blocks so I wrote my own remark plugin and want to share with the community

https://nextra-theme-docs-dl1oz9i5p.vercel.sh/docs/advanced/mermaid

### CODE REVIEW

File: packages/nextra/package.json [hunk 1]
```diff
@@ -86,9 +90,15 @@
     "slash": "^3.0.0"
   },
   "peerDependencies": {
+    "mermaid": ">=9.1.5",
     "next": ">=9.5.3",
     "react": ">=16.13.1",
-    "react-dom": ">=18.2.0"
+    "react-dom": ">=16.13.1"
```
Comment (B2o5T) [inline packages/nextra/package.json:96]: mistake, peerDeps `react-dom` must be same as `react`!

File: packages/nextra/package.json [hunk 2]
```diff
@@ -86,9 +90,15 @@
...
+  },
+  "peerDependenciesMeta": {
+    "mermaid": {
+      "optional": true
+    }
```
Comment (B2o5T) [inline packages/nextra/package.json:101]: making this package optional so users that don't use mermaid will not download this package

File: packages/nextra/src/mdx-plugins/static-image.js [hunk 3]
```diff
@@ -1,57 +1,28 @@
...
+export const remarkStaticImage = () => (tree, _file, done) => {
     const importsToInject = []
 
     visit(tree, 'image', visitor)
-    tree.children.unshift(...importsToInject)
-    tree.children.unshift(ASTNodeImport('$NextImageNextra', 'next/image'))
+    tree.children.unshift(
+      getASTNodeImport('$NextImageNextra', 'next/image'),
+      ...importsToInject
+    )
```
Comment (B2o5T) [inline packages/nextra/src/mdx-plugins/static-image.js:21]: https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1865

File: packages/nextra/src/mermaid.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+let id = 0
+
+export default function Mermaid({ chart }: { chart: string }): ReactElement {
+  const [svg, setSVG] = useState('')
+
+  useEffect(() => {
+    mermaid.initialize({
+      startOnLoad: true,
+      theme:
+        typeof localStorage === 'undefined' ? undefined : localStorage.theme
```
Comment (B2o5T) [inline packages/nextra/src/mermaid.tsx:16]: any through on how to access `useTheme` hook inside this component and use `theme` value here?

**Comment by shuding** [comment]
I think it will be definitely helpful for the community to publish it as a standalone plugin btw :)

File: packages/nextra/src/mermaid.tsx (hunk 4 above)
Comment (shuding) [reply packages/nextra/src/mermaid.tsx:16]: I think you can directly use `useTheme` in this component

**Pull request closed without merging**